### DISCUSSION TITLE
Done Button in Whatsapp mode

### DISCUSSION THREAD

**Discussion by leandro-pilz** [hm21/pro_image_editor#59] [Ideas]
Hello, as I mentioned, we really like this WhatsApp theme, but I would like to see with you the possibility of having the done button at the top, because when we are not using the text field with the send button next to it, we would not be able to save the changes, this would be one of the ways to use it.   The other would be with the text field and the lower send button that you already have in the examples.

**Comment by hm21** [comment]
Just to confirm, you want to have the Done button at the top right where the arrow is in the image below:

![image](https://github.com/hm21/pro_image_editor/assets/40503456/a716aebc-5d08-4b68-ae94-78222f6d1341)

Technically, I can also add a customWidget option where you can add it. But the problem with the WhatsApp design is that there are already many buttons in the AppBar, which means on smaller devices like a Galaxy Fold you will get a problem with overflow. However, if you know the devices of your users, and they don't use tiny devices, I will add it for you after you confirm that the top right position is correct.

**Comment by leandro-pilz** [comment]
Perfect, that's right, and maybe you could remove the stikers option there,
sorry for the inconvenience, but anyway, or have the option of not using
it, which for us having textEditor, Paint and crop/rotate is perfect, well
I sent one email a little ago with the issue of the context bug when using
asuka, in this version the bug continues.

Em ter., 28 de mai. de 2024 às 16:18, Alex Frei ***@***.***>
escreveu:

**Comment by hm21** [comment]
Okay, I released the new version `3.0.0-dev.21` which allows you to add your own done button, and you can also enable/disable only the subeditors you need. Below is an example how you can remove the emojis/stickers and with a custom done button.

```dart
configs: ProImageEditorConfigs(
  emojiEditorConfigs: const EmojiEditorConfigs(enabled: false),
  // Remove this configs => stickerEditorConfigs: StickerEditorConfigs(...),
  customWidgets: ImageEditorCustomWidgets(
    whatsAppOwnAppBarIcons: Padding(
      padding: const EdgeInsets.only(left: 10.0),
      child: IconButton(
        tooltip: 'Done',
        onPressed: () {
          editorKey.currentState?.doneEditing();
        },
        icon: const Icon(Icons.check),
        style: whatsAppButtonStyle,
      ),
    ),
  ),
```
